tests/Oryx.Tests.Common/EndToEndTestHelper.cs

```diff
@@ -20,7 +20,7 @@ namespace Microsoft.Oryx.Tests.Common
     public static class EndToEndTestHelper
     {
         private const int MaxRetryCount = 100;
-        private static readonly TimeSpan DelayBetweenRetries = TimeSpan.FromSeconds(100);
+        private static readonly TimeSpan DelayBetweenRetries = TimeSpan.FromSeconds(10);
```

Comment:
@william-msft was there a reason that we needed to delay a full 100 seconds AND retry 100 times? I thought that checking for liveliness every 10 seconds, for a total of ~16 minutes would be enough to prevent timeouts, and it saves us time on the tests that I've tried it with (since we check for the first time after 10 seconds instead of a full 100).

Reply:
Ok so the nightly tests have answered this for me with a suite of failures. I still have some questions regarding this wait, namely, why are we waiting so long before checking for container existence? This wait seems to me like it could be adding significant time to our testing suite.

I think a better approach would be to:
1. Initiate container run
2. Check for container existence every few seconds for a certain period of time, if it doesn't exist after ~2-3 minutes then the test fails
3. Then we begin checking for the app to be up every few seconds for a certain period of time

It seems like this would also separate the 2 processes of checking for the container to exist and the app to startup. It also would allow for containers that are from cached images and don't need to be pulled down (so they will start up much faster) to not need to be waiting for a full 100 seconds before even checking for the application to be up.

Happy to give this a shot, but let me know if its already been explored at all or any thoughts you have for it!

Reply:
The above approach sounds good to me -- as long as there is no performance hit or downside to checking the container existence and application on a faster interval than we should go ahead with it.